src/enclave/mod.rs

```diff
@@ -15,3 +15,100 @@ mod ioctls;
 
 pub use builder::*;
 pub use enclave::*;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::types::page;
+    use crate::types::tcs::Tcs;
+
+    use std::num::NonZeroU32;
+
+    use lset::Span;
+    use primordial::Page;
+
+    #[allow(clippy::erasing_op)]
+    const CODE_OFFSET: usize = Page::size() * 0;
+    const TCS_OFFSET: usize = Page::size() * 1;
+    const SSA_OFFSET: usize = Page::size() * 2;
+    const SSA_COUNT: NonZeroU32 = unsafe { NonZeroU32::new_unchecked(1) };
+
+    /// Don't call this function. It has a non-standard ABI.
+    #[naked]
+    unsafe extern "C" fn code() -> ! {
+        asm!(
+            "0:",
+            "inc rdi", // Increment all the registers.
+            "inc rsi",
+            "inc rdx",
+            "inc r8",
+            "inc r9",
+            "xchg rbx, rcx", // Get the exit address in RBX.
+            "mov rax, 0x04", // Set RAX = EEXIT.
+            "enclu",         // Do the exit.
+            "1:",
+            ".fill((4096 - (1b - 0b)))",
+            options(noreturn)
+        )
+    }
+
+    #[test]
+    #[cfg_attr(not(has_sgx), ignore)]
+    fn test() {
+        let span = Span {
+            start: 1 * 1024 * 1024 * 1024,
+            count: 1 * 1024 * 1024,
+        };
+
+        let mut builder = Builder::new(span).unwrap();
+
+        let segs: [Segment; 3] = [
+            Segment {
+                src: vec![unsafe { *(code as *const Page) }],
```

Comment:
There should definitely be a comment explaining why you're doing this. Also maybe an apology.